### PR TITLE
Improved autocomplete suggestions

### PR DESCRIPTION
**Pull request by jerminatorhits**
## Description
Why did you write this code?
swap react-google-maps SearchBox with self google api component since the library doesn't allow in-depth customizations for location

## How to Test
- [x] Visit '/' or dev03.beenest.io
- [x] Type inside input and see cities suggested
- [x] Verify all query params are still valid and being sent on submit

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?

## Learnings
Did you learn anything here you want to share with the team?

`react-google-maps` doesn't expose very many options to customize as beenest grows. It's good for small/side projects so as our product grows, it may be good to slowly move away.

### CODE REVIEW

File: src/components/shared/GoogleAutoComplete/GoogleAutoComplete.tsx [hunk 1]
```diff
@@ -11,30 +10,50 @@ import { SETTINGS } from 'configs/settings';
...
+  autocomplete: AutocompleteInterface;
+  
+  componentDidMount() {
+    if (!this.props.inputRef.current) return;
 
-class StandaloneSearchBoxWithErrorBoundary extends React.Component<BoundaryProps> {
-  state = { hasError: false };
+    this.autocomplete = new google.maps.places.Autocomplete(
+      this.props.inputRef.current,
+      {"types": ["(cities)"]}
```
Comment (tc) [inline src/components/shared/GoogleAutoComplete/GoogleAutoComplete.tsx]: does using the google maps api directly let us prioritize recommended cities? ie, typing in `new` should show `new york` 

I did notice on dev03 typing in `new` showed `newark, new york` vs some random city on beenest.com

Comment (jerminatorhits) [reply src/components/shared/GoogleAutoComplete/GoogleAutoComplete.tsx]: I think only these types can be used https://developers.google.com/places/supported_types#table3

I'll do a bit more research to see it can be adjusted further. If it can't be changed, another autocomplete library could be an option.

Comment (tc) [reply src/components/shared/GoogleAutoComplete/GoogleAutoComplete.tsx]: we should support cities (new york) and regions (hawaii)

**Comment by tc** [comment]
@kcvan +1 it's in a better state than we found it.
